### ISSUE TITLE
incomplete format string in _get_element_and_line()

### ISSUE THREAD

**Issue by jonnyprouty** [hyperspy/hyperspy#2448]
[hyperspy/misc/eds/utils.py](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/misc/eds/utils.py) contains an error in _get_element_and_line(). Specifically, ValueError is passed an incomplete format string. The following code demonstrates the issue:
```
s = hs.signals.Signal1D(np.random.random((10,100)))
s.set_signal_type("EDS_SEM")
s.get_lines_intensity("Ca_Kb") # should raise KeyError

```
Returns the following:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-14-9a0d450cf112> in <module>
      1 s = hs.signals.Signal1D(np.random.random((10,100)))
      2 s.set_signal_type("EDS_SEM")
----> 3 s.get_lines_intensity("Ca_Kb") # should raise KeyError

~/.local/anaconda3/lib/python3.7/site-packages/hyperspy/_signals/eds.py in get_lines_intensity(self, xray_lines, integration_windows, background_windows, plot_result, only_one, only_lines, **kwargs)
    618         """
    619 
--> 620         xray_lines = self._parse_xray_lines(xray_lines, only_one, only_lines)
    621         if hasattr(integration_windows, '__iter__') is False:
    622             integration_windows = self.estimate_integration_windows(

~/.local/anaconda3/lib/python3.7/site-packages/hyperspy/_signals/eds.py in _parse_xray_lines(self, xray_lines, only_one, only_lines)
    523                                           only_lines=only_lines)
    524         xray_lines, xray_not_here = self._get_xray_lines_in_spectral_range(
--> 525             xray_lines)
    526         for xray in xray_not_here:
    527             warnings.warn("%s is not in the data energy range." % xray +

~/.local/anaconda3/lib/python3.7/site-packages/hyperspy/_signals/eds.py in _get_xray_lines_in_spectral_range(self, xray_lines)
    156         xray_lines_not_in_range = []
    157         for xray_line in xray_lines:
--> 158             line_energy = self._get_line_energy(xray_line)
    159             if low_value < line_energy < high_value:
    160                 xray_lines_in_range.append(xray_line)

~/.local/anaconda3/lib/python3.7/site-packages/hyperspy/_signals/eds.py in _get_line_energy(self, Xray_line, FWHM_MnKa)
     87                     "`set_signal_type(\"EDS_SEM\")` to convert to one of these"
     88                     "signal types.")
---> 89         line_energy = utils_eds._get_energy_xray_line(Xray_line)
     90         if units_name == 'eV':
     91             line_energy *= 1000

~/.local/anaconda3/lib/python3.7/site-packages/hyperspy/misc/eds/utils.py in _get_energy_xray_line(xray_line)
     30     By example, if xray_line = 'Mn_Ka' this function returns 5.8987
     31     """
---> 32     element, line = _get_element_and_line(xray_line)
     33     return elements_db[element]['Atomic_properties']['Xray_lines'][
     34         line]['energy (keV)']

~/.local/anaconda3/lib/python3.7/site-packages/hyperspy/misc/eds/utils.py in _get_element_and_line(xray_line)
     20     lim = xray_line.find('_')
     21     if lim == -1:
---> 22         raise ValueError("Invalid xray-line: %" % xray_line)
     23     return xray_line[:lim], xray_line[lim + 1:]
     24 

ValueError: incomplete format
```

**Comment by francisco-dlp** [comment]
@jonnyprouty, thank you for the bug report.

**Comment by ericpre** [comment]
The issue is that this function is expecting an iterable https://github.com/hyperspy/hyperspy/blob/fe7602e4ff9ae51a86c2a4c0d0545971ef972ae3/hyperspy/_signals/eds.py#L553 and not a str and the error message is not appropriate. The fix would be to raise a `TypeError` with an appropriate error message or convert the string to a list of string.
@jonnyprouty, are you interested in trying to contribute by making a pull request fixing this issue?

**Comment by jonnyprouty** [comment]
>thank you for the bug report.

You're welcome, @francisco-dlp.

> are you interested in trying to contribute by making a pull request fixing this issue?

I'd be happy to do that, @ericpre. I should be able to take care of it tomorrow.

**Comment by ericpre** [comment]
Great, we have a developer guide: https://hyperspy.org/hyperspy-doc/current/dev_guide/intro.html, in case you haven't come across it yet!

**Comment by tjof2** [comment]
Closed in #2450?
